relay-server/src/envelope.rs

```diff
@@ -710,20 +710,31 @@ impl Item {
     }
 
     /// Sets the payload and content-type of this envelope.
-    pub fn set_payload<B>(&mut self, content_type: ContentType, payload: B)
-    where
+    pub(crate) fn set_payload_optional_content_type<B>(
+        &mut self,
+        content_type: Option<ContentType>,
+        payload: B,
+    ) where
         B: Into<Bytes>,
     {
         let mut payload = payload.into();
 
-        let length = std::cmp::min(u32::max_value() as usize, payload.len());
+        let length = std::cmp::min(u32::MAX as usize, payload.len());
         payload.truncate(length);
 
         self.headers.length = Some(length as u32);
-        self.headers.content_type = Some(content_type);
+        self.headers.content_type = content_type;
         self.payload = payload;
     }
 
+    /// The primary payload setter requires a content-type, since most call-sites set it explicitly.
```

Comment:
The original docstring is more descriptive:
```suggestion
    /// Sets the payload and content-type of this envelope.
```